### PR TITLE
LPD-25641 Submit for Workflow an Document with Expire date

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
I'm working over `LPD-25788` and `LPD-25640` so 'on-hold' till the PRs are merged and maybe a rebase is needed.

Steps to Test

1. modify : System Settings > Documents and Media > SYSTEM SCOPE > service : check interval to 1 (to test purposes)
2. Crate a Folder
3. Edit the folder to put single approver as workflow option
4. Create File with name “one” without expiration date on the "folder" with expiration date “now + 2 min”
5. Wait 5 minutes
6. Go to workflow assignments and approve it.

> When the document has an expiration date set previous to the date when the reviewer approves the document, the document will be set to [Approved] directly and the expiration date will be removed, checking the Never expire checkbox

Other :

1. modify : System Settings > Documents and Media > SYSTEM SCOPE > service : check interval to 1 (to test purposes)
2. Crate a Folder
3. Edit the folder to put single approver as workflow option
4. Create File with name “one” without expiration date on the "folder" with expiration date “now + 1 year”
5. Wait 5 minutes
6. Go to workflow assignments and approve it.

> When the document has an expiration date set after the date when the reviewer approves the document, the document will be set to [Approved] and the expiration date will remain, being the document set to [Expired] once the expiration date is reached.

### commits to check on this PR
- **LPD-25641 refactor code**
- **LPD-25641 add test to check the behavior when we approve a document with an expiration date**
- **LPD-25641 When the document has an expiration date set previous to the date when the reviewer approves the document, the document will be set to Approved directly and the expiration date will be removed**

### CODE REVIEW

**Comment by AliciaGarciaGarcia** [comment]
ci:test:content-management

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

File: modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/internal/workflow/test/DLFileEntryWorkflowHandlerTest.java [hunk 1]
```diff
@@ -61,70 +61,104 @@ public class DLFileEntryWorkflowHandlerTest {
 	@Before
 	public void setUp() throws Exception {
 		_group = GroupTestUtil.addGroup();
+
+		_serviceContext = ServiceContextTestUtil.getServiceContext(
+			_group.getGroupId(), TestPropsValues.getUserId());
+
+		_folder = _getSingleApproverFolder();
```
Comment (jkappler) [inline modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/internal/workflow/test/DLFileEntryWorkflowHandlerTest.java:68]: @AliciaGarciaGarcia, all looks good :). Here I would just pass the serviceContext as parameter to explicitly note that an order should be followed.

What do you think?

Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/internal/workflow/test/DLFileEntryWorkflowHandlerTest.java:68]: great idea, I'll do it

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:content-management

**Comment by AliciaGarciaGarcia** [comment]
@jkappler 
rebased and sent with the proposed changes

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:report:2572311591

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:content-management

**Comment by AliciaGarciaGarcia** [comment]
ci:report:2572616158

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:report:2572690295

**Comment by jkappler** [comment]
ci:report:2572744178

**Comment by jkappler** [comment]
@AliciaGarciaGarcia , don't see any unique failures related to your changes. So I think it's safe to move forward

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:report:2573158361

**Comment by AliciaGarciaGarcia** [comment]
no unique test failures
no common test failures seems to be related with PR

**Comment by AliciaGarciaGarcia** [comment]
ci:forward:force

**Pull request closed without merging**